### ISSUE TITLE
EOM doesn't rotate Exif thumbnail

### ISSUE THREAD

**Issue by cykerway** [mate-desktop/eom#173]
#### Expected behaviour

-   Image is rotated 90 degree clockwise.
-   Exif thumbnail is rotated 90 degree clockwise.

#### Actual behaviour

-   Image is rotated 90 degree clockwise.
-   But Exif thumbnail stays the same.

#### Steps to reproduce the behaviour

1.  Shoot a picture using a mobile device.
2.  Open the image with EOM.
3.  Click *Rotate Clockwise* button.
4.  Save the rotated image.

#### MATE general version

N/A

#### Package version

1.20.0

#### Linux Distribution

Arch Linux

#### Link to downstream report of your Distribution

N/A

**Comment by raveit65** [comment]
Eom Gallery shows me the rotated thumbnail.
Same with thumbnail in properties window.
What do you mean?

**Comment by cykerway** [comment]
I mean exactly what I said in **Steps to reproduce the behaviour**.

Basically, I took a picture with my phone, copied to my PC. Then I extracted its thumbnail using:

    exiv2 -et a.jpg

Then I rotated it with EOM and extracted thumbnail again. Now I have 2 thumbnails. But they are identical files. Even the sha1sum are the same. I was expecting the second thumbnail to be rotated.

The thumbnail is rotated when I use another program to rotate the image:

    exiftran -9 -i a.jpg

So I think the problem may be with EOM. Then I looked into its source and found it seems to have taken care of thumbnail rotation. This confused me. I'm using EOM 1.20.0.

**Comment by raveit65** [comment]
> I mean exactly what I said in Steps to reproduce the behaviour.

> #### Steps to reproduce the behaviour

> 1.  Shoot a picture using a mobile device.
> 2.  Open the image with EOM.
> 3.  Click *Rotate Clockwise* button.
> 4.  Save the rotated image.

You didn't say that you extract the thumbnail with exif in first post :)

Any way do you have a example file with exif data ?
```
exiv2 -et bridge-at-sunset-from-riga-3840x2160_79784-mm-90-test.jpg 
bridge-at-sunset-from-riga-3840x2160_79784-mm-90-test.jpg: No Exif data found in the file
```
Btw. the thumbnail in caja file browser is rotated, after i saved the image in eom after rotation.

**Comment by cykerway** [comment]
>   Btw. the thumbnail in caja file browser is rotated, after i saved the image in eom after rotation.

Yes, very true. The same thing happens with my caja as well. I haven't looked into caja, but caja probably doesn't use the Exif thumbnail. Try test using pcmanfm (which prefers the Exif thumbnail when compiled with Exif support) or extract the Exif thumbnail using `exiv2` or `exiftool`.

**Comment by cykerway** [comment]
For your convenience, the Exif thumbnail extraction commands are here:

    exiv2 -et a.jpg
    exiftool -b -ThumbnailImage a.jpg > a-thumb.jpg

**Comment by raveit65** [comment]
See, my post i need an image with exif data for reproducing.
I like to live under the radar, so i don't use technologies with gps sensors ;)

**Comment by raveit65** [comment]
Hmm, using
```
exiftool -b -ThumbnailImage bridge-at-sunset-from-riga-3840x2160_79784-mm-90-test.jpg > bridge-at-sunset-from-riga-3840x2160_79784-mm-90-test-2.jpg
```
works, but i can't open the result in eom/eog, and caja shows me no image.

**Comment by cykerway** [comment]
As I said, you can grab any image file and manually add a Exif thumbnail to it:

    exiftran -i -g a.jpg

The `exiftran` tool is part of a package named `fbida`. So you may want to search for that name instead in your package manager (if you use a Linux distro). [fbida-2.13](https://www.kraxel.org/releases/fbida/fbida-2.13.tar.gz) works on my machine.

**Comment by raveit65** [comment]
Using the  -g option ((re)generate thumbnail) gives me a segfault.
```
[rave@mother test]$ exiftran -g -i a.jpg 
processing a.jpg
Segmentation fault (core dumped)
[rave@mother test]$ exiftran -g  a.jpg -o b.jpg
Segmentation fault (core dumped)
```

**Comment by cykerway** [comment]
Are you using version 2.13? I have segfault with 2.14, but 2.13 works fine.

Additionally, if you go to that download page, they have even older versions that may worth a try if 2.13 doesn't work.

If none of them works, then you probably have to take out your mobile and take a pic.

**Comment by raveit65** [comment]
Again, i don't use a cell phone with gps.
Error message is with 2.14 from fedora.
2.13 doesn't compile because of non existent dependency.
```
No match for argument: pkgconfig(cairo-gl)
Error: Unable to find a match
make: *** [GNUmakefile:31: yum] Error 1
```
Fedora doesn't have a package cairo-gl
Sorry, i can't help you if you don't send me a file which have a exif thumbnail.

**Comment by cykerway** [comment]
OK Brother. Let's see if [this](https://image.ibb.co/nwhY2T/a.jpg) works.

**Comment by raveit65** [comment]
Ok, that works and i can confirm that the resulting exif-thumb isn't rotated.
Eye of gnome has the same behavior, but it works if i rotate the image with gimp.

**Comment by cykerway** [comment]
Thank you for confirming this. I spent some time on it before I realized the problem might be within the image editor. I also reviewed [eom-image.c](https://github.com/mate-desktop/eom/blob/v1.20.0/src/eom-image.c#L441) which seems to give thumbnail rotation a consideration. But the output is still wrong. Hope this helps your debugging.

**Comment by raveit65** [comment]
@cykerway  
Looks like this is fixed with eom-1.22?
I can't reproduce it with your image any more.

**Comment by cykerway** [comment]
1.22.2 here but no, still can't rotate thumbnail...

**Comment by raveit65** [comment]
Weird, here in fedora 30and eom-1.22 the thumbnails are rotated.
![Bildschirmfoto zu 2019-12-09 17-05-10](https://user-images.githubusercontent.com/961604/70451801-54ee0600-1aa6-11ea-8fa1-a9018d21c2af.png)

What do i wrong?

**Comment by cykerway** [comment]
I didn't read its code and don't know whether that's actually a thumbnail or not. You can use [this online tool](http://exif.regex.info/exif.cgi) to extract its thumbnail. I tested with GIMP and it works correctly.
